Assets/Scripts/Command/ChangeHierarchyOrder.cs

```diff
@@ -0,0 +1,174 @@
+using System;
+using Assets.Scripts.Events;
+using Assets.Scripts.SceneState;
+using UnityEngine;
+
+namespace Assets.Scripts.Command
+{
+    public class ChangeHierarchyOrder : SceneState.Command
+    {
+        private readonly Guid? draggedEntityId;
+        private readonly Guid? onTopEntityId;
+        private readonly Guid? aboveEntityId;
+        private readonly Guid? belowEntityId;
+
+        private DclEntity startPrimarySelection;
+        private readonly DclEntity startParent;
+        private readonly float? startHierarchyOrder;
+        private readonly Vector3 startPosition;
+        private readonly Vector3 startScale;
+        private readonly Quaternion startRotation;
+        private readonly Placement placement;
+
+        public ChangeHierarchyOrder(DclEntity draggedEntity, DclEntity onTopEntity, DclEntity aboveEntity,
```

Comment:
But at that point it's not a parent/sibling/child yet, and it would be confusing to call it a parent, because the entity you're placing the dragged entity on, doesn't necessarily have any children.